Documentation/media/v4l-drivers/bcm2835-isp.rst

```diff
@@ -0,0 +1,118 @@
+.. SPDX-License-Identifier: GPL-2.0
+
+BCM2835 ISP Driver
+==================
+
+Introduction
+------------
+
+The BCM2835 Image Sensor Pipeline (ISP) is a fixed function hardware pipeline
+for performing image processing operations. Input frames (in Bayer format) may
+be obtained from the BCM2835 Unicam peripheral, written to a memory framebuffer,
+and fed into the input of the ISP.  Two output images may be generated (in
+YUV or RGB format), along with image statistics.
+
+The bcm2835-isp driver exposes the following media pads as V4L2 device nodes:
+
+.. tabularcolumns:: |l|l|l|l|
+
+.. cssclass: longtable
+
+.. flat-table::
+
+    * - *Pad*
+      - *Direction*
+      - *Purpose*
+      - *Formats*
+
+    * - "bcm2835-isp0-output0"
+      - sink
+      - Accepts Bayer format frame buffers as input to the ISP HW pipeline.
+      - :ref:`RAW8 <V4L2-PIX-FMT-SRGGB8>`,
+        :ref:`RAW10 <V4L2-PIX-FMT-SRGGB10>`,
+        :ref:`RAW10P <V4L2-PIX-FMT-SRGGB10P>`,
+        :ref:`RAW12 <V4L2-PIX-FMT-SRGGB12>`,
+        :ref:`RAW12P <V4L2-PIX-FMT-SRGGB12P>`,
+        :ref:`RAW14 <V4L2-PIX-FMT-SRGGB14P>`,
+        :ref:`RAW16 <V4L2-PIX-FMT-SRGGB16>`.
+
+    * - "bcm2835-isp0-capture1"
+      - source
+      - High resolution YUV or RGB processed output from the ISP.
+      - :ref:`YUV420 <V4L2-PIX-FMT-YUV420>`,
+        :ref:`NV12/NV12 <V4L2-PIX-FMT-NV12>`,
+        :ref:`RGB565 <V4L2-PIX-FMT-RGB565>`,
+        :ref:`RGB24/BGR24 <V4L2-PIX-FMT-RGB24>`,
+        :ref:`ABGR32 <V4L2-PIX-FMT-ABGR32>`,
```

Comment:
FYI I've just had a forum query as to why we do ABGR but not ARGB on the codecs (mainly encoder in). 
The main reason was that there wasn't a matching VC_IMAGE_T type, so it wasn't a quick mapping.

For encoder and ISP input it's input formatter config, so I'll look at adding it.
ISP output should be able to support other ordering too (I think it's only those two, not the other arrangements of the 4 components).